scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_model.py

```diff
@@ -249,46 +409,312 @@ def change_plot_guess(self, plot_guess, parameter_dict):
+
+    def get_runs_groups_and_pairs_for_fits(self):
+        selected_workspaces = self.get_selected_workspace_list()
+        runs = []
+        groups_and_pairs = []
+
+        if self.fit_type == "Single":
+            for workspace in selected_workspaces:
+                runs += [get_run_number_from_workspace_name(workspace, self.context.data_context.instrument)]
+                groups_and_pairs += [get_group_or_pair_from_name(workspace)]
+            run_groups_and_pairs = sorted(zip(runs, groups_and_pairs))
+            groups_and_pairs = [grp_pair for _, grp_pair in run_groups_and_pairs]
+            runs = [run for run, _ in run_groups_and_pairs]
+        else:
+            fit_workspaces = {}
+            if self.fit_by == "Run":
+                for workspace in selected_workspaces:
+                    run = get_run_number_from_workspace_name(workspace, self.context.data_context.instrument)
+                    if run not in fit_workspaces:
+                        fit_workspaces[run] = get_group_or_pair_from_name(workspace)
+                    else:
+                        fit_workspaces[run] += ";" + get_group_or_pair_from_name(workspace)
+                runs = list(fit_workspaces.keys())
+                runs.sort()
+                groups_and_pairs = list(fit_workspaces.values())
+            elif self.fit_by == "Group/Pair":
+                for workspace in selected_workspaces:
+                    group_or_pair = get_group_or_pair_from_name(workspace)
+                    run = get_run_number_from_workspace_name(workspace, self.context.data_context.instrument)
+                    if group_or_pair not in fit_workspaces:
+                        fit_workspaces[group_or_pair] = run
+                    else:
+                        fit_workspaces[group_or_pair] += ";" + run
+                runs = list(fit_workspaces.values())
+                groups_and_pairs = list(fit_workspaces.keys())
+            else:
+                return
+
+        return runs, groups_and_pairs
+
+    def get_fit_workspace_names_from_groups_and_runs(self, runs, groups_and_pairs):
+        workspace_names = []
+        for run in runs:
+            for group_or_pair in groups_and_pairs:
+                if group_or_pair in self.context.group_pair_context.selected_pairs:
+                    workspace_names += [get_pair_asymmetry_name(self.context, group_or_pair, run, not self.fit_to_raw)]
+                else:
+                    workspace_names += [get_group_asymmetry_name(self.context, group_or_pair, run, not self.fit_to_raw)]
+
+        return workspace_names
+
+    def workspace_list_sorter(self, workspace_name):
+        run_number = get_run_number_from_workspace_name(workspace_name, self.context.data_context.instrument)
+        grp_pair_number = self._transform_grp_or_pair_to_float(workspace_name)
+        return int(run_number) + grp_pair_number
+
+    def _transform_grp_or_pair_to_float(self, workspace_name):
```

Comment:
can we have a comment explaining what this function is for and why it returns 0.999?